src/tasks/installers/operator.ts

```diff
@@ -60,6 +61,66 @@ export class OperatorTasks {
           }
         }
       },
+      {
+        title: 'Checking for pre-created TLS secret',
+        // In case of Openshift infrastructure the certificate from cluster router will be used, so no need in the `che-tls` secret.
+        skip: () => !isKubernetesPlatformFamily(flags.platform),
+        task: async (_: any, task: any) => {
+          // Che is being deployed on Kubernetes infrastructure
+
+          if (!(flags.tls || await this.checkCrForTls(flags))) {
+            // No TLS mode, skip this check
+            return
+          }
+
+          const cheSecretName = 'che-tls'
+          const cheSecret = await kube.getSecret(cheSecretName, flags.chenamespace)
+          if (cheSecret) {
+            task.title = `${task.title}... "${cheSecretName}" secret found`
+            return
+          }
+
+          // The secret is required but doesn't exist, show error message.
+          const errorMessage =
+            `Che TLS mode is turned on, but required "${cheSecretName}" secret is not pre-created in "${flags.chenamespace}" namespace, so Eclipse Che cannot be started. \n` +
+            'This is not bug in Eclipse Che and such behavior is expected. \n' +
+            'Please refer to Che documentation for more informations: ' +
+            'https://www.eclipse.org/che/docs/che-7/setup-che-in-tls-mode-with-self-signed-certificate/'
+          throw new Error(errorMessage)
+        }
+      },
+      {
+        title: 'Checking certificate',
+        // If the flag is set no need to check if it is required
+        skip: () => flags['self-signed-cert'],
+        task: async (_: any, task: any) => {
+          if (!(flags.tls || await this.checkCrForTls(flags))) {
+            // No TLS mode, skip this check
+            return
+          }
+
+          const warningMessage = 'Self-signed certificate is used, so "--self-signed-cert" option is required. Added automatically.'
+
+          const platform = flags.platform
+          if (platform === 'minikube' || platform === 'crc' || platform === 'minishift') {
```

Comment:
QE and from time to time I use `https://console-openshift-console.apps.ocp43.codereadyqe.com/` where self-singed certs are used. And AFAIU it's not covered in this PR, so as before I will get broken Che if run `chectl server:start --installer=operator --platform=openshift` :disappointed: 
If it's not easy to determinate - let's ask users if cluster uses self-signed certs or not, like
```
TLS is used by default, does https://console-openshift-console.apps.ocp43.codereadyqe.com/ use self-signed certificate? Y/n
...
```
```

Reply:
Thank you for deep review! Yes, you are right, however I had mentioned that as advanced case in the PR description, maybe not clear enough, sorry.
I may add this prompt, but I am not sure it is right way to go. It would bring another problems: how it would work in case of automated installation, say on PR check? new flag to disable this check? I think we have to implement certificate verification, which is separate issue. Until that users have to read docs as it was decided on planning...

Reply:
>  Yes, you are right, however I had mentioned that as advanced case in the PR description, maybe not clear enough, sorry.

I saw it but still I imagine my emotions again and again when I remember your PR after figuring out that my Che is broken and I must reinstall it :cry: 

> It would bring another problems: how it would work in case of automated installation, say on PR check? new flag to disable this check? 

Ask that only if we enable tls by default
If user specifies `--tls` then we can blame themselves in missing `--self-signed-cert` :smiling_imp:

Reply:
I'll add a warning in case when it is not clear if self-signed certificate used.

Reply:
> Ask that only if we enable tls by default

So always

Reply:
I know that we have uncovered that case but it is not trivial and it would be better to handle it in a separate PR.